test/integration/node/lifecycle_test.go

```diff
@@ -45,6 +50,301 @@ const poll = 2 * time.Second
+			nodeIndex := 1
+			nodeCount := 3
+			var nodes []*v1.Node
+			for i := 0; i < nodeCount; i++ {
+				node := &v1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:   fmt.Sprintf("testnode-%d", i),
+						Labels: map[string]string{"node.kubernetes.io/exclude-disruption": "true"},
+					},
+					Spec: v1.NodeSpec{},
+					Status: v1.NodeStatus{
+						Conditions: []v1.NodeCondition{
+							{
+								Type:   v1.NodeReady,
+								Status: v1.ConditionTrue,
+							},
+						},
+					},
+				}
+				nodes = append(nodes, node)
+			}
+			testPod := &v1.Pod{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:       "testpod",
+					Finalizers: []string{"foo.bar/some-finalizer"},
+				},
+				Spec: v1.PodSpec{
+					NodeName: nodes[nodeIndex].Name,
+					Containers: []v1.Container{
+						{Name: "container", Image: imageutils.GetPauseImageName()},
+					},
+				},
+				Status: v1.PodStatus{
+					Phase: v1.PodRunning,
+					Conditions: []v1.PodCondition{
+						{
+							Type:   v1.PodReady,
+							Status: v1.ConditionTrue,
+						},
+					},
+				},
+			}
+
+			defer featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, features.JobBackoffPolicy, test.enableJobBackoffPolicy)()
+			testCtx := testutils.InitTestAPIServer(t, "with-job-backoff-policy", nil)
+
+			// Build clientset and informers for controllers.
+			testCtx = testutils.InitTestScheduler(t, testCtx)
+			defer testutils.CleanupTest(t, testCtx)
+			cs := testCtx.ClientSet
+
+			// Build clientset and informers for controllers.
+			externalClientConfig := restclient.CopyConfig(testCtx.KubeConfig)
+			externalClientConfig.QPS = -1
+			externalClientset := kubernetes.NewForConfigOrDie(externalClientConfig)
+			externalInformers := informers.NewSharedInformerFactory(externalClientset, time.Second)
+
+			// Start NodeLifecycleController for taint.
+			nc, err := nodelifecycle.NewNodeLifecycleController(
+				testCtx.Ctx,
+				externalInformers.Coordination().V1().Leases(),
+				externalInformers.Core().V1().Pods(),
+				externalInformers.Core().V1().Nodes(),
+				externalInformers.Apps().V1().DaemonSets(),
+				cs,
+				5*time.Second,    // Node monitor grace period
+				time.Minute,      // Node startup grace period
+				time.Millisecond, // Node monitor period
+				1,                // Pod eviction timeout
+				100,              // Eviction limiter QPS
+				100,              // Secondary eviction limiter QPS
+				50,               // Large cluster threshold
+				0.55,             // Unhealthy zone threshold
+				true,             // Run taint manager
+			)
+			if err != nil {
+				t.Errorf("Failed to create node controller: %v", err)
+				return
+			}
+
+			// Waiting for all controllers to sync
+			externalInformers.Start(testCtx.Ctx.Done())
+			externalInformers.WaitForCacheSync(testCtx.Ctx.Done())
+			testutils.SyncInformerFactory(testCtx)
+
+			// Run all controllers
+			go nc.Run(testCtx.Ctx)
+			go testCtx.Scheduler.Run(testCtx.Ctx)
```

Comment:
Good point, I have eliminated the scheduler and the test works. The test was based on the "TestTaintBasedEvictions" which uses Scheduler, but it might be unnecessary there too.